src/main/java/com/minecolonies/coremod/colony/buildings/BuildingShepherd.java

```diff
@@ -0,0 +1,106 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.client.gui.WindowHutWorkerPlaceholder;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.jobs.AbstractJob;
+import com.minecolonies.coremod.colony.jobs.JobShepherd;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Creates a new building for the blacksmith.
+ */
+public class BuildingShepherd extends AbstractBuildingWorker
+{
+    /**
+     * Description of the job executed in the hut.
+     */
+    private static final String SHEPHERD          = "Shepherd";
+
+    /**
+     * Description of the block used to set this block.
+     */
+    private static final String SHEPHERD_HUT_NAME = "shepherdHut";
+
+    /**
+     * Max building level of the hut.
+     */
+    private static final int MAX_BUILDING_LEVEL = 5;
+
+    /**
+     * Instantiates the building.
+     * @param c the colony.
+     * @param l the location.
+     */
+    public BuildingShepherd(final Colony c, final BlockPos l)
+    {
+        super(c, l);
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return SHEPHERD;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
+
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return SHEPHERD;
+    }
+
+    //TODO Implement Later
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)